api/data_explorer/__main__.py

```diff
@@ -282,6 +285,22 @@ def _process_facets():
     es_facets = OrderedDict()
     ui_facets = OrderedDict()
 
+    # Add a 'Samples Overview' facet if sample_file_columns were specified in
+    # bigquery.json.
+    if app.app.config['SAMPLE_FILE_COLUMNS']:
+        filters_facet = FiltersFacet({
+            'Has %s' % key:
+            Match(**{'samples._has_%s' % key.lower().replace(' ', '_'): True})
+            for key in app.app.config['SAMPLE_FILE_COLUMNS'].keys()
+        })
+        es_facets['Samples Overview'] = NestedFacet('samples', filters_facet)
+        ui_facets['Samples Overview'] = {
+            # TODO(bryancrampton): This needs to be generalized to support filtering
+            # across multiple fields, and to support the _has_* sample file fields.
+            'elasticsearch_field_name': 'samples',
```

Comment:
FYI - I think the export SQL is the only thing elasticsearch_field_name is used for. (As I noted in another comment, export SQL is broken)